das/dastree/dastree.go

```diff
@@ -43,26 +49,44 @@ func RecordHash(record func(bytes32, []byte), preimage ...[]byte) bytes32 {
 	//  Intermediate hashes like '*' from above may be recorded via the `record` closure
 	//
 
-	keccord := func(value []byte) bytes32 {
+	keccord := func(value []byte) (bytes32, error) {
 		hash := crypto.Keccak256Hash(value)
-		record(hash, value)
-		return hash
+		err := record(hash, value)
+		return hash, err
```

Comment:
Could maybe be
```go
return hash, record(hash, value)
```

Reply:
Reverted the changes to the record signature.